In [34]:
import pandas as pd

In [35]:
df_sports = pd.read_csv('/Users/kinga/Documents/DataScience/Task/sports.csv', names=['number', 'customer_id', 'sport'], header=0)
df_sports.head()

number  customer_id       sport
0       1   77461851.0    p3ywanie
1       2   37934116.0   myolistwo
2       3   39770481.0  3ucznictwo
3       4   50298765.0  nurkowanie
4       5   23383753.0  3ucznictwo

In [36]:
df_sports.info

<bound method DataFrame.info of           number  customer_id        sport
0              1   77461851.0     p3ywanie
1              2   37934116.0    myolistwo
2              3   39770481.0   3ucznictwo
3              4   50298765.0   nurkowanie
4              5   23383753.0   3ucznictwo
...          ...          ...          ...
1189742  1189743   70183840.0   nurkowanie
1189743  1189744   39250750.0  jeYdziectwo
1189744  1189745   22997615.0    myolistwo
1189745  1189746   52840571.0   nurkowanie
1189746  1189747          NaN          NaN

[1189747 rows x 3 columns]>

In [37]:
df_sports.isna().sum()

number         0
customer_id    1
sport          1
dtype: int64

In [38]:
df_sports_cleaned = df_sports.dropna(axis=0, how='any')

Removing NaN values

In [39]:
df_sports_cleaned = df_sports_cleaned.drop(df_sports_cleaned.columns[0], axis=1)

Removing the column "number", there is a DataFrame index

In [40]:
sport_types = df_sports_cleaned['sport'].unique()
sport_types

array(['p3ywanie', 'myolistwo', '3ucznictwo', 'nurkowanie', 'bieganie',
       'jeYdziectwo', 'rowery', 'myslistwo', 'karate', 'aquafitness',
       'chodzenie', 'judo', 'hokej', 'snowboard'], dtype=object)

Checking the number of unique values in "sport"


In [41]:
df_sports_cleaned.loc[df_sports_cleaned['sport'] == 'myolistwo', 'sport'] = 'myslistwo'

In [42]:
df_sports_cleaned.loc[df_sports_cleaned['sport'] == '3ucznictwo', 'sport'] = 'lucznictwo'

In [43]:
df_sports_cleaned.loc[df_sports_cleaned['sport'] == 'jeYdziectwo', 'sport'] = 'jezdziectwo'

In [44]:
df_sports_cleaned.loc[df_sports_cleaned['sport'] == 'p3ywanie', 'sport'] = 'plywanie'


Cleaning the sport names

In [46]:
df_sports_cleaned['sport'].value_counts(sort=True, ascending=True)

sport
hokej           21409
chodzenie       21423
snowboard       21472
judo            21476
aquafitness     21566
karate          21695
rowery          43038
bieganie        53111
jezdziectwo     63640
myslistwo      191489
nurkowanie     211743
lucznictwo     222829
plywanie       274855
Name: count, dtype: int64

The least popular sport is hockey (hokej)

In [47]:
n_o_sports = df_sports_cleaned['customer_id'].value_counts()
n_o_sports

customer_id
58372354.0    3
31293700.0    3
73689399.0    3
25310262.0    3
37389217.0    3
             ..
4198067.0     1
10856715.0    1
8091621.0     1
50883872.0    1
14246106.0    1
Name: count, Length: 1060705, dtype: int64

In [48]:
multiple_sports = n_o_sports[n_o_sports > 2]
multiple_sports.count()

np.int64(5040)

There are 5040 clients who practice more than two sports